src/main/groovy/com/google/protobuf/gradle/KtDslCompatibilityUtils.java

```diff
@@ -0,0 +1,34 @@
+package com.google.protobuf.gradle;
+
+import kotlin.Unit;
+import kotlin.jvm.functions.Function1;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.kotlin.dsl.NamedDomainObjectContainerExtensionsKt;
+import org.gradle.kotlin.dsl.NamedDomainObjectContainerScope;
+
+public class KtDslCompatibilityUtils {
+
+    /**
+     * This method is used to delegate the NamedDomainObjectContainer configuration to
+     * to which ever kotlin-dsl ext implementation is available on the classpath.
+     *
+     * Since NamedDomainObjectContainerExtensionsKt.invoke is an inline extension function, our
+     * usages in 'ProtobufConfiguratorExts.kt' would use the byte code for which ever kotlin-dsl
+     * version we compiled against. This caused issues with providing compatibility with
+     * kotlin-dsl 1.0.0-rc3 and 1.0.4 (Stable).
+     *
+     * Since the kotlin compiler creates a static implementations of all extensions, we can create a
+     * delegating utility function that ensures we are not inline-ing our kotlin-dsl compilation
+     * target byte code.
```

Comment:
Sorry, I still don't understand why this additional layer of indirection would prevent the problem. Doesn't this method inline the `invoke()` call too?